### PR TITLE
ShearBook environment and config

### PR DESCRIPTION
**Pull request by sfarrens**
## Summary 

- Pinned PyPi/Conda package versions in `environment.yml`
- Dropped GalSim from conda as this is a dependency of Piff
- Updated `_config.yml` with latest myst standards

### CODE REVIEW

**Comment by aboucaud** [comment]
So I had the issue with Justine that GalSim is not available on conda-forge for windows users.
Removing it from the `environment.yml` is certainly good.
I'm just wondering if it just moves the problem or solves it to go through Piff (which probably installs Galsim through pip and not conda).

**Comment by sfarrens** [comment]
So Piff does install Galsim from PyPi. Personally, I prefer the Conda build as it tends to be more stable, but the Conda and PyPi versions do not get along so it has to be one or the other. So, if we include Piff in the environment, we have to drop the Conda build.

One thing we can probably do to make life easier for Windows users is make a Docker image with everything pre-installed. Then anyone can build the book locally and run the Jupyter using a Docker container as the backend.

**Comment by aboucaud** [comment]
Yep, good point. Now that we have GitHub registry it's a good way of interacting with the book.
Would you like me to have a look at it or are you planning to do it ?

**Comment by aboucaud** [comment]
So here's what I found 
- [ ] you should list `pip` as a conda dependency in order to avoid a ton of warning messages (new behaviour) but perhaps without pinning it
- [ ] I would suggest pinning the Python version to 3.8 because some of the libs may not be well compatible with 3.9
- [ ] Axel would like the env name to be renamed to `shear-book` to avoid the confusion with possible other envs

That's it !

**Comment by aboucaud** [comment]
Otherwise it works !

**Comment by sfarrens** [comment]
Just testing my Dockerfile, if everything works I will push the new commits.

File: environment.yml [hunk 1]
```diff
@@ -3,11 +3,12 @@ channels:
...
 dependencies:
-  - jupyter
-  - matplotlib
-  - numpy
-  - scipy
-  - galsim
+  - jupyter==1.0.0
+  - matplotlib==3.4.2
+  - numpy==1.21.0
+  - scipy==1.7.0
```
Comment (aboucaud) [inline environment.yml:9]:
```suggestion
  - scipy==1.7.0
  - pip
```

File: environment.yml [hunk 2]
```diff
@@ -3,11 +3,12 @@ channels:
   - conda-forge
   - defaults
 dependencies:
-  - jupyter
-  - matplotlib
-  - numpy
-  - scipy
-  - galsim
+  - jupyter==1.0.0
```
Comment (aboucaud) [inline environment.yml:6]:
```suggestion
  - python=3.8
  - jupyter==1.0.0
```

File: environment.yml (hunk 1 above)
Comment (aboucaud) [inline environment.yml]:
```suggestion
  - python=3.8
  - jupyter=1.0.0
  - matplotlib=3.4.2
  - numpy=1.21.0
  - scipy=1.7.0
  - pip
```

**Review by aboucaud — COMMENTED** [review]
One should also update the `requirements.txt` accordingly.

File: environment.yml (hunk 1 above)
Comment (sfarrens) [reply environment.yml]: Included in latest commit.

**Comment by sfarrens** [comment]
> One should also update the `requirements.txt` accordingly.

I just removed it....

... but now I need to update the CI accordingly.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM continuumio/miniconda3
+
+LABEL Description="ShearBook Docker Image"
+WORKDIR /home
+ENV SHELL /bin/bash
+
+RUN git clone https://github.com/CosmoStat/Shear-and-PSF-Reading-Group.git
```
Comment (aboucaud) [inline Dockerfile:7]: Don't you want to build the `develop` branch for now ?

Comment (aboucaud) [reply Dockerfile:7]: Sorry, read too quickly

**Comment by aboucaud** [comment]
Why would you want to remove it ?
I do not use `conda` and I usually find it easier to have a `requirements.txt` to peak at the dependencies of a given repo.

File: environment.yml (hunk 1 above)
Comment (aboucaud) [reply environment.yml]: conda syntax uses single `=` unlike pip. But it should work the way it is.

**Comment by sfarrens** [comment]
> Why would you want to remove it ?
> I do not use `conda` and I usually find it easier to have a `requirements.txt` to peak at the dependencies of a given repo.

OK, I can re-add it. I removed it because it's easier to keep everything consistent if package versions are only specified in one place.

**Comment by aboucaud** [comment]
My last comment sounded a bit harsh :) sorry about that.
I like the idea of improving consistency and removing files that can go against smooth maintenance.

That said, I do feel like the requirement.txt is a must have in any Python repo, no matter the other available options.

**Comment by aboucaud** [comment]
I guess you forgot the file :D

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+- jupyter=1.0.0
+- matplotlib=3.4.2
+- numpy=1.21.0
+- scipy=1.7.0
+- jupyter-book=0.11.2
+- piff=1.0.0
+- sep=1.2.0
+- sympy=1.8
```
Comment (aboucaud) [inline requirements.txt]:
```suggestion
- jupyter==1.0.0
- matplotlib==3.4.2
- numpy==1.21.0
- scipy==1.7.0
- jupyter-book==0.11.2
- piff==1.0.0
- sep==1.2.0
- sympy==1.8
```
😉

Comment (sfarrens) [reply requirements.txt]: Too slow! 😄

**Comment by sfarrens** [comment]
@aboucaud OK, everything should be OK now. Let me know if you spot anything else.

File: .github/workflows/test_build.yml [hunk 5]
```diff
@@ -13,17 +13,14 @@ jobs:
...
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          python -m pip install -r requirements.txt
-          python -m pip install .
+          activate-environment: shearbook
+          environment-file: environment.yml
 
       - name: Build Book
+        shell: bash -l {0}
```
Comment (aboucaud) [inline .github/workflows/test_build.yml:23]: Just curious, why was that necessary ?

**Review by aboucaud — DISMISSED** [review]
LGTM 👍🏼

File: .github/workflows/test_build.yml (hunk 5 above)
Comment (sfarrens) [reply .github/workflows/test_build.yml:23]: This is required for this step to see the active environment set up in the previous step. I guess if you don't put this a new independent shell is launched for each step.

Comment (aboucaud) [reply .github/workflows/test_build.yml:23]: Waow, really not obvious. 😳

**Review by aboucaud — APPROVED** [review]

**Comment by aboucaud** [comment]
Shall I merge ?

**Comment by sfarrens** [comment]
Hi @aboucaud, sure go ahead. :)
